docs/tutorials/Containers.md

```diff
@@ -0,0 +1,475 @@
+
+```wdl
+task hello_world {
+    String name = "World"
+    command {
+        echo 'Hello, ${name}'
+    }
+    output {
+        File out = stdout()
+    }
+    runtime {
+        docker: 'ubuntu:latest'
+    }
+}
+
+workflow hello {
+    call hello_world
+}
+```
+
+Similarly in CWL, you can specify a [`DockerRequirement`](https://www.commonwl.org/v1.0/CommandLineTool.html#DockerRequirement) inside the requirements section:
+
+```cwl
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: echo
+inputs:
+    name:
+        type: string
+        default: "World"
+        inputBinding:
+          prefix: "Hello, "
+outputs:
+    out: stdout
+
+requirements:
+    DockerRequirement:
+        dockerPull: "ubuntu:latest"
+``` 
+___
+### Docker
+
+[Docker](https://www.docker.com) is a popular container technology that is natively supported by Cromwell and WDL.
+
+
+#### Docker on a Local Backend
+
+On a single machine (laptop or server), no extra configuration is needed to allow docker to run, provided Docker is installed.
+
+You can install Docker for Linux, Mac or Windows from [Docker Hub](https://hub.docker.com/search/?type=edition&offering=community)
+
+#### Docker on Cloud
+
+It is strongly advised that you provide a Docker image to tasks that will run on Cloud backends, and in fact most Cloud providers require it.
+
+It might be possible to use an alternative container engine, but this is not recommended if Docker is supported.
+
+#### Docker on HPC
+
+Docker can allow running users to gain superuser privileges, called the [Docker daemon attack surface](https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface). In HPC and multi-user environments, Docker recommends that "only trusted users should be allowed to control your Docker Daemon".
+
+For this reason it's worth exploring other technologies that will support the reproducibility and simplicity of running a workflow that uses docker containers; Singularity or udocker.
+
+___
+
+### Singularity
+
+Singularity is a container engine designed for use on HPC systems in particular, while ensuring an appropriate level of security that Docker cannot provide.
+
+#### Installation
+Before you can configure Cromwell on your HPC system, you (or your sysadmin) will have to install Singularity, which is documented [here](https://www.sylabs.io/guides/3.0/admin-guide/admin_quickstart.html#installation).
+Installation of Singularity doesn't require root, but to access the full features of Singularity, such as the use of its preferred image format, the Singularity binary must be owned by root, with the `setuid` bit enabled.
+This is [documented here](https://www.sylabs.io/guides/2.6/admin-guide/security.html#how-does-singularity-do-it).
+If you do not give Singularity these privileges, you are restricted to using sandbox images, which come with a number of downsides.
```

Comment:
I wouldn't encourage using Sandbox images either. The entire point is to ensure reproducibility, and with suggestions like this, we are undoing that. We should encourage the user to build read only squash fs binaries.

Reply:
I don't think I'm trying to encourage the use of sandboxes here, it's just that it's the only option without `setuid` on `singularity`